Detector/DetFCChhTrackerSimple/src/GenericBarrel_Geo.cpp

```diff
@@ -0,0 +1,154 @@
+         ++xCompColl, ++idxSubMod) {
+      DD4hep::XML::Component xComp = static_cast<DD4hep::XML::Component>(xCompColl);
+      std::string moduleComponentName = "layer" + std::to_string(idxLay) + "_rod_module_component" +
+          std::to_string(idxSubMod) + "_" + xComp.materialStr();
+      Volume moduleComponentVolume(moduleComponentName,
+                                   DD4hep::Geometry::Box(xModule.width(), xComp.thickness(), xModule.length()),
+                                   lcdd.material(xComp.materialStr()));
+      moduleComponentVolume.setVisAttributes(lcdd, xComp.visStr());
+      if (xComp.isSensitive()) {
+        moduleComponentVolume.setSensitiveDetector(sensDet);
+      }
+      moduleComponentVector.push_back(moduleComponentVolume);
+    }
+
+    // definition of module volume (smallest independent subdetector)
+    // define the module whose name was given in the "layer" xml Element
+    Volume moduleVolume("module", DD4hep::Geometry::Box(xModule.width(), xModule.thickness(), xModule.length()),
+                        lcdd.material("Air"));
+    moduleVolume.setVisAttributes(lcdd, xModule.visStr());
+
+    // definition of rod volume (longitudinal arrangement of modules)
+    Volume rodVolume("GenericTrackerBarrel_layer" + std::to_string(idxLay) + "_rod",
+                     DD4hep::Geometry::Box(xModule.width(), xModule.thickness(), xLayer.dz()),
+                     lcdd.material("Air"));
+    rodVolume.setVisAttributes(lcdd.invisible());
+
+    /// @todo: allow for more than one type of module components
+    // analogous to module
+    // place module substructure in module
+    std::string moduleComponentName = "moduleComponent";
+    idxSubMod = 0;
+    for (DD4hep::XML::Collection_t xCompColl(xModuleComponents, _U(module_component)); nullptr != xCompColl;
+         ++xCompColl, ++idxSubMod) {
+      DD4hep::XML::Component xComp = static_cast<DD4hep::XML::Component>(xCompColl);
+      DD4hep::Geometry::Position offset(0, -0.5 * totalModuleComponentThickness + integratedModuleComponentThickness,
+                                        0);
+      integratedModuleComponentThickness += xComp.thickness();
+      PlacedVolume placedModuleComponentVolume = moduleVolume.placeVolume(moduleComponentVector[idxSubMod], offset);
+      placedModuleComponentVolume.addPhysVolID("module_component", idxSubMod);
+    }
+
+    double layerThickness;
+    unsigned int numRepeat;
+    double layer_rmin;
+    // handle repeat attribute in xml
+    // "repeat" layers  equidistant between rmin and rmax
+    numRepeat = xLayer.repeat();
+    layerThickness = (xLayer.rmax() - xLayer.rmin()) / numRepeat;
+    layer_rmin = xLayer.rmin();
+    unsigned int nPhi = 0;
+    double r = 0;
+    double phi = 0;
+    // loop over repeated layers defined by one layer tag
+    for (unsigned int repeatIndex = 0; repeatIndex < numRepeat; ++repeatIndex, ++idxLay) {
```

Comment:
In that case please move that ++idxLay into the body of the loop. That makes the semantics more clear.